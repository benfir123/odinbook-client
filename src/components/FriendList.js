import {
  Avatar,
  List,
  ListItem,
  ListSubheader,
  ListItemIcon,
  ListItemText,
  Divider,
  IconButton,
  ListItemSecondaryAction,
} from "@mui/material";
import CheckCircleIcon from "@mui/icons-material/CheckCircle";
import CancelIcon from "@mui/icons-material/Cancel";

const FriendList = ({
  friends,
  friendRequests,
  handleAcceptRequest,
  handleDeclineRequest,
}) => {
  return (
    <div>
      <List
        component="nav"
        aria-label="friend requests list"
        subheader={<ListSubheader>Friend Requests</ListSubheader>}
      >
        <div>
          {friendRequests.length > 0
            ? friendRequests.map((friend) => {
                return (
                  <ListItem key={friend._id}>
                    <ListItemIcon>
                      <Avatar
                        alt={friend.full_name}
                        src={friend.profile_pic_url}
                      />
                    </ListItemIcon>
                    <ListItemText primary={friend.full_name} />
                    <ListItemSecondaryAction>
                      <IconButton
                        edge="end"
                        aria-label="comments"
                        onClick={() => handleAcceptRequest(friend._id)}
                      >
                        <CheckCircleIcon sx={{ color: "#15803d" }} />
                      </IconButton>
                      <IconButton
                        edge="end"
                        aria-label="comments"
                        onClick={() => handleDeclineRequest(friend._id)}
                      >
                        <CancelIcon sx={{ color: "#991b1b" }} />
                      </IconButton>
                    </ListItemSecondaryAction>
                  </ListItem>
                );
              })
            : "No friend requests yet"}
          <Divider />
        </div>
      </List>
      <List
        component="nav"
        aria-label="[top] friends list"
        subheader={<ListSubheader>Friends</ListSubheader>}
      >
        <div>
          {friends.length > 0
            ? friends.map((friend) => {
                return (
                  <ListItem key={friend._id}>
                    <ListItemIcon>
                      <Avatar
                        alt={friend.full_name}
                        src={friend.profile_pic_url}
                      />
                    </ListItemIcon>
                    <ListItemText primary={friend.full_name} />
                  </ListItem>
                );
              })
            : "No friends yet"}
          <Divider />
        </div>
      </List>
    </div>
  );
};

export default FriendList;
