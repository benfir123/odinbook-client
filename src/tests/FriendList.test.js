import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import FriendList from "../components/FriendList";

describe("FriendList component", () => {
  it("renders the correct elements", () => {
    const { container } = render(
      <FriendList
        friends={[
          {
            _id: 1,
            full_name: "Ben Chanapai",
            profile_pic_url: "",
          },
          {
            _id: 2,
            full_name: "John Chanapai",
            profile_pic_url: "",
          },
        ]}
        friendRequests={[
          {
            _id: 1,
            full_name: "Ben Chanapai",
            profile_pic_url: "",
          },
          {
            _id: 2,
            full_name: "John Chanapai",
            profile_pic_url: "",
          },
        ]}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
