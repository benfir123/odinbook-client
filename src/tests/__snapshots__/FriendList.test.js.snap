// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FriendList component renders the correct elements 1`] = `
<div>
  <div>
    <nav
      aria-label="friend requests list"
      class="MuiList-root MuiList-padding MuiList-subheader css-p64g7a-MuiList-root"
    >
      <li
        class="MuiListSubheader-root MuiListSubheader-gutters MuiListSubheader-sticky css-uob957-MuiListSubheader-root"
      >
        Friend Requests
      </li>
      <div>
        <li
          class="MuiListItem-container css-oyakqy-MuiListItem-container"
        >
          <div
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-secondaryAction css-hcmkxp-MuiListItem-root"
          >
            <div
              class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
                  data-testid="PersonIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                Ben Chanapai
              </span>
            </div>
          </div>
          <div
            class="MuiListItemSecondaryAction-root css-518kzi-MuiListItemSecondaryAction-root"
          >
            <button
              aria-label="comments"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1qz8dnu-MuiSvgIcon-root"
                data-testid="CheckCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="comments"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1n7txvp-MuiSvgIcon-root"
                data-testid="CancelIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </li>
        <li
          class="MuiListItem-container css-oyakqy-MuiListItem-container"
        >
          <div
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-secondaryAction css-hcmkxp-MuiListItem-root"
          >
            <div
              class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
                  data-testid="PersonIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                John Chanapai
              </span>
            </div>
          </div>
          <div
            class="MuiListItemSecondaryAction-root css-518kzi-MuiListItemSecondaryAction-root"
          >
            <button
              aria-label="comments"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1qz8dnu-MuiSvgIcon-root"
                data-testid="CheckCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="comments"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1n7txvp-MuiSvgIcon-root"
                data-testid="CancelIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </li>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
      </div>
    </nav>
    <nav
      aria-label="[top] friends list"
      class="MuiList-root MuiList-padding MuiList-subheader css-p64g7a-MuiList-root"
    >
      <li
        class="MuiListSubheader-root MuiListSubheader-gutters MuiListSubheader-sticky css-uob957-MuiListSubheader-root"
      >
        Friends
      </li>
      <div>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
        >
          <div
            class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
                data-testid="PersonIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiListItemText-root css-tlelie-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
            >
              Ben Chanapai
            </span>
          </div>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
        >
          <div
            class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
                data-testid="PersonIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiListItemText-root css-tlelie-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
            >
              John Chanapai
            </span>
          </div>
        </li>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
      </div>
    </nav>
  </div>
</div>
`;
